hudi-spark-datasource/hudi-spark2/src/main/scala/org/apache/spark/sql/adapter/Spark2Adapter.scala

```diff
@@ -122,6 +123,30 @@ class Spark2Adapter extends SparkAdapter {
     InterpretedPredicate.create(e)
   }
 
+  override def createHoodieFileScanRDD(sparkSession: SparkSession,
+                                       readFunction: PartitionedFile => Iterator[InternalRow],
+                                       filePartitions: Seq[FilePartition],
+                                       readDataSchema: StructType,
+                                       metadataColumns: Seq[AttributeReference] = Seq.empty): FileScanRDD = {
+    new Spark2HoodieFileScanRDD(sparkSession, readFunction, filePartitions)
+  }
+
+  override def resolveDeleteFromTable(deleteFromTable: Command,
```

Comment:
Given that we have `extractCondition` we can get rid of `resolveDeleteFromTable`

Reply:
We can't simply reuse the method, `resolveDeleteFromTable`  has a different logic that involves `resolveFromExpression`

Reply:
I don't see why we can't: 

- We get rid of the method completely 
- We use `extractCondition` to extract condition and then do everything else (resolution, etc) in the caller